tests/0113-cooperative_rebalance.cpp

```diff
@@ -30,83 +30,306 @@
+                                 std::vector<RdKafka::TopicPartition *> toppars2) {
+  RdKafka::ErrorCode err;
+  RdKafka::Error *error;
+  std::vector<RdKafka::TopicPartition *> assignment;
+
+  test_assert(assignment.size() == 0, "Expecting current assignment to have size 0");
+  if ((err = consumer->assign(toppars1))) Test::Fail("Assign failed: " + RdKafka::err2str(err));
+  if ((err = consumer->assignment(assignment))) Test::Fail("Failed to get current assignment: " + RdKafka::err2str(err));
+  test_assert(assignment.size() == 1, "Expecting current assignment to have size 1");
+  delete assignment[0];
+  assignment.clear();
+  if ((error = consumer->incremental_unassign(toppars1))) {
+    Test::Fail("Incremental unassign failed: " + error->str());
+    delete error;
+  }
+  if ((err = consumer->assignment(assignment))) Test::Fail("Failed to get current assignment: " + RdKafka::err2str(err));
+  test_assert(assignment.size() == 0, "Expecting current assignment to have size 0");
+}
+
+/** incremental assign, then incremental unassign
+ */
+static void direct_assign_test_3(RdKafka::KafkaConsumer *consumer,
+                                 std::vector<RdKafka::TopicPartition *> toppars1,
+                                 std::vector<RdKafka::TopicPartition *> toppars2) {
+  RdKafka::ErrorCode err;
+  RdKafka::Error *error;
+  std::vector<RdKafka::TopicPartition *> assignment;
+
+  test_assert(assignment.size() == 0, "Expecting current assignment to have size 0");
+  if ((error = consumer->incremental_assign(toppars1))) {
+    Test::Fail("Incremental assign failed: " + error->str());
+    delete error;
+  }
+  if ((err = consumer->assignment(assignment))) Test::Fail("Failed to get current assignment: " + RdKafka::err2str(err));
+  test_assert(assignment.size() == 1, "Expecting current assignment to have size 1");
+  delete assignment[0];
+  assignment.clear();
+  if ((error = consumer->incremental_unassign(toppars1))) {
+    Test::Fail("Incremental unassign failed: " + error->str());
+    delete error;
+  }
+  if ((err = consumer->assignment(assignment))) Test::Fail("Failed to get current assignment: " + RdKafka::err2str(err));
+  test_assert(assignment.size() == 0, "Expecting current assignment to have size 0");
+}
+
+/** multi-topic incremental assign and unassign + message consumption.
+ */
+static void direct_assign_test_4(RdKafka::KafkaConsumer *consumer,
+                                 std::vector<RdKafka::TopicPartition *> toppars1,
+                                 std::vector<RdKafka::TopicPartition *> toppars2) {
+  std::vector<RdKafka::TopicPartition *> assignment;
+
+  consumer->incremental_assign(toppars1);
+  consumer->assignment(assignment);
+  test_assert(assignment.size() == 1, "Expecting current assignment to have size 1");
+  delete assignment[0];
+  assignment.clear();
+  RdKafka::Message *m = consumer->consume(5000);
+  test_assert(m->err() == RdKafka::ErrorCode::ERR_NO_ERROR, "Expecting a consumed message.");
```

Comment:
`RdKafka::ERR_NO_ERROR`, for whatever reason.

Reply:
oh yeah, C++'s notion of an enum is a bit stupid.